### ISSUE TITLE
`AzureAD\` Prefix is ignored and you have to manually type it every time you want to login via RDP

### ISSUE THREAD

**Issue by BrainSlugs83** [mRemoteNG/mRemoteNG#1846]
It appears as though the `AzureAD\` user principal name prefix is not honored as specified.  This causes login to fail on many versions of windows (See Recording below) when that machine is joined to an Azure AD domain (as opposed to an on-prem or hybrid AD domain).

Here is a recording of this behavior, where you can see that I have specified `AzureAD\` as a prefix to my user name (in the bottom left), and that the login fails and I have to manually type it to login (you can click on the image to see a full screen version):

![AzureAD-Domain-Broken](https://user-images.githubusercontent.com/5217366/91517877-bad42500-e8a3-11ea-8a8f-7e4ce32dfb76.gif)

## Expected Behavior
If the user types in `AzureAD\user@domain.com` as their user name, it should actually treat that as their VERBATIM user name.  The `AzureAD\` prefix should not be treated as a domain, it's a hint to tell windows that you're using AzureAD, and on some builds of windows it is a required "hint" if you want to login remotely. 

(Additionally, I can verify that it also does not work if you specify it as your domain instead of as a prefix -- either way, it appears to be ignored.)

## Current Behavior
`AzureAD` is ignored, no matter what field you type it into, it is never sent across the wire, and so automated login is not possible on some builds of windows and windows server.

## Possible Solution
Again, if the user types `AzureAD\` as a prefix for their user name, mRemoteNG should honor that and send it across as the full verbatim user name. -- I have used the FreeRDP source in another project to automate logging in to remote desktops, and I know that this solution works.  If you send the full user principal name with that prefix, the login will succeed.

## Steps to Reproduce (for bugs)
1.  Create connections in mRemoteNG to remote desktops that are AzureAD domain joined (not on-prem).
2.  Notice that some of them will fail to login with the stored credentials in mRemoteNG (I believe anything after 1903?), because it does not honor the `AzureAD` prefix.
3.  Note that no matter where you specify the `AzureAD` prefix (either in the domain field or the user name field) that it is ignored, and that you will have to type it *again* at the login screen.

## Context
I am trying to log in to multiple remote desktops at once, and I would like be able to store my credentials so that I do not have to type them in 25 different times.

## Your Environment

* Version used: 1.76.20.24615
* Operating System and version (e.g. Windows 10 1709 x64): Windows 10 x64 (1903 / 18362.1016), Windows 10 x64 (1909 / 18363.1016), Windows 10 x64 (2004 / 19041.450)

**Comment by farosch** [comment]
***AzureAD\user@domain.com*** is **not** a username.
Username is **AzureAD\user**
Domain is **domain.com**
Put those information in the appropriate fields and things should work:
![image](https://user-images.githubusercontent.com/22905106/93265020-486eaa80-f7a8-11ea-8ec4-bc23d0e55961.png)


I don't have any way to test this out but I cannot think of any reason why this shouldn't - If not feel free to reopen this for further investigation.

**Comment by BrainSlugs83** [comment]
That does not work. -- Here is the error message you see if you try it that way.

![image](https://user-images.githubusercontent.com/5217366/96930338-634fd100-1470-11eb-91f6-9fe30a27aac1.png)

**Comment by BrainSlugs83** [comment]
@farosch  -- I don't have an option to reopen, is it something you can do?

**Comment by BrainSlugs83** [comment]
At any rate regardless of whether `AzureAD\user@domain.com` is a username or not -- it's what you have to put in the username field.

**Comment by matt-psaltis** [comment]
For myself (in a year's time) and anyone else who might benefit:

```
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Terminal Server\WinStations\RDP-Tcp
Change "SecurityLayer" to 0
```

Disable CredSSP (Set to No) in connection profile
Username: `.\AzureAD\<upn>` (yes the dot slash is required to avoid the AzureAD prefix from being removed)
Password: <Office 365 password>
Domain: <blank>

**Comment by farosch** [comment]
@BrainSlugs83 Sry for being absent so long!
I will take a look at this.

**Comment by nrjohnstone** [comment]
@matt-psaltis great find for the work around... it also works without disabling CredSSP for anyone else wanting to connect to their Azure VMs

**Comment by jbenson-montgomery-twochicago** [comment]
> For myself (in a year's time) and anyone else who might benefit:
> 
> ```
> HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Terminal Server\WinStations\RDP-Tcp
> Change "SecurityLayer" to 0
> ```
> 
> Disable CredSSP (Set to No) in connection profile Username: `.\AzureAD\<upn>` (yes the dot slash is required to avoid the AzureAD prefix from being removed) Password: `<Office 365 password>` Domain: `<blank>`

This was the answer for me as well, although the registry change wasn't required. Thank you so much for the tip!

**Comment by dllimoges** [comment]
Yes, Out of the box still does not work. The "SecurityLayer" was already zero. 
I did get RDP to work, but NLA must be off on the AzureAD machine OR AzureAD VM and had to add these 2 lines to the end of the RDP file:

enablecredsspsupport:i:0
authentication level:i:2

(Worked like a champ, RDP does pass ".\AzureAD\")

The Question, how to add these 2 lines to mRemoteNG to see if it will help?
Turning off CredSSP creates a new error "The connection cannot proceed because authentication is not enabled and the remote computer requires authentication to be enabled to connect."

If I blank out the Username & Password, manually enter the creds, it times out.

Just wondering if anybody has figured this out.

**Comment by TiloGit** [comment]
> For myself (in a year's time) and anyone else who might benefit:
> 
> ```
> HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Terminal Server\WinStations\RDP-Tcp
> Change "SecurityLayer" to 0
> ```
> 
> Disable CredSSP (Set to No) in connection profile Username: `.\AzureAD\<upn>` (yes the dot slash is required to avoid the AzureAD prefix from being removed) Password: `<Office 365 password>` Domain: `<blank>`

PS for that
```powershell
## READ THE VALUE
Get-Itemproperty -path "HKLM:\SYSTEM\CurrentControlSet\Control\Terminal Server\WinStations\RDP-Tcp" -Name 'SecurityLayer'
## Set the Value
Set-Itemproperty -path "HKLM:\SYSTEM\CurrentControlSet\Control\Terminal Server\WinStations\RDP-Tcp" -Name 'SecurityLayer' -value 0
```

if anybody wonder that the values are: 
https://learn.microsoft.com/en-us/windows-hardware/customize/desktop/unattend/microsoft-windows-terminalservices-rdp-winstationextensions-securitylayer

<html><body>


0 | Specifies that the Microsoft Remote Desktop Protocol (RDP) is  used by the server and the client for authentication before a remote  desktop connection is established. RDP is a Microsoft protocol that  supports terminal services across heterogeneous network environments.
-- | --
1 | Specifies that the server and the client negotiate the method for  authentication before a remote desktop connection is established. This  is the default value.
2 | Specifies that the Transport Layer Security (TLS) protocol is  used by the server and the client for authentication before a remote  desktop connection is established.


</body>
</html>

**Comment by Nasyncs** [comment]
For my future self:

Username: domain\username
Password: O365 Password
Domain: Blank
Use CredSSP: No

On the laptop side, under System properties>remote:
Check "Allow Remote Assistance connections to this computer"
Only select "Allow remote connection to this computer", no need to hit the check box underneath for NLA. 


This allows me to connect with any other modifications.

**Comment by ageisen2000** [comment]
Just throwing in my "fix"

Username: .\AzureAD\myname@mydomain.com
Password: My O365 password
Domain: Blank

Use CredSSP: No
Server Authentication: "Warn me if authentication fails"

The server authentication part was what was missing for me.

**Comment by robertpopa22** [comment]
**Fix available for testing**

A fix for this issue has been implemented in our community fork (`robertpopa22/mRemoteNG`, branch `main`). A beta build including this fix is available from the fork's Releases page.

Please test and report if this resolves your issue.

---

> *We sincerely appreciate the mRemoteNG maintainers and the wonderful community that has kept this project alive and thriving for over a decade. Your work powers thousands of IT professionals worldwide, including us. Our fork builds entirely on your foundation — we're preparing a consolidated Pull Request to contribute these improvements back to the official project. Thank you for everything you've built.*
>
> *— robertpopa22, community contributor*
